### PR TITLE
Allow detekt to run on kts too.

### LINKED ISSUE #612: Allow checks on Kotlin script files (.kts)
Right now, a check for a `.kt` extension is [hard coded](https://github.com/arturbosch/detekt/blob/33b4ee1d249d0845d75099108caea136a125560f/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/KtTreeCompiler.kt#L47) in detekt.

detekt should be able to check Kotlin script files as well (`.kts` extension).

### PR DESCRIPTION
**Pull request by vanniktech**
Fixes #612

Let's hope it'll be that easy

### CODE REVIEW

File: detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/KtTreeCompiler.kt [hunk 1]
```diff
@@ -39,7 +39,7 @@ class KtTreeCompiler(private val compiler: KtCompiler = KtCompiler(),
 	private fun Path.isKotlinFile(): Boolean {
 		val fullPath = this.toAbsolutePath().toString()
 		val kotlinEnding = fullPath.substring(fullPath.lastIndexOf('.') + 1)
-		return kotlinEnding.length == 2 && kotlinEnding.endsWith("kt")
+		return kotlinEnding == "kt" || kotlinEnding == "kts"
```
Comment (arturbosch) [inline detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/KtTreeCompiler.kt]: Please test for `.kt` and `.kts` or files like `config.detekt` will get analyzed too ^^

Comment (arturbosch) [reply detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/KtTreeCompiler.kt]: oh sorry, my fault the kotlinEnding line does the split

**Comment by arturbosch** [comment]
Please add a test case where a kotlin script is successfully compiled.

**Comment by vanniktech** [comment]
Sure, where's the place for that?

**Comment by arturbosch** [comment]
Ok, the KtTreeCompiler does the `isKotlinFile` check so the test case should be in `detekt-core/KtTreeCompilerSpec`. In `resources/cases` you can out a kt script file. Furthermore please change this line `project.isFile() -> listOf(compiler.compile(project, project))` to 
`project.isFile() && project.isKotlinFile() -> listOf(compiler.compile(project, project))`. Now you can use the `compile(scriptPath)` function to test it ^^

**Comment by vanniktech** [comment]
`it("should work with two or more filters") {` fails now. Any hints?

**Comment by arturbosch** [comment]
Well the script name is `KotlinScript.kts` and does not match one of the filters. I must admit the resource structure is not optimal for independent tests ....

**Pull request merged**